app/helpers/admin/orders_helper.rb

```diff
@@ -7,7 +7,17 @@ module OrdersHelper
     # We exclude shipping method adjustments because they are displayed in a
     # separate table together with the order line items.
     def order_adjustments_for_display(order)
-      order.adjustments + order.all_adjustments.payment_fee.eligible
+      adjustments_for_display = order.adjustments + order.all_adjustments.payment_fee.eligible
+
+      if VoucherAdjustmentsService.new(order).voucher_included_tax.negative?
+        adjustment = order.voucher_adjustments.first
+        adjustments_for_display << Spree::Adjustment.new(
+          label: I18n.t("admin.orders.edit.voucher_tax_included_in_price", label: adjustment.label),
+          amount: adjustment.included_tax
+        )
+      end
+
+      adjustments_for_display
```

Comment:
The whole `if` block could have been somewhere else, e.g. the service and then this could be simpler:
```suggestion
      voucher_service = VoucherAdjustmentsService.new(order)

      order.adjustments +
        order.all_adjustments.payment_fee.eligible +
        voucher_service.included_tax_adjustment
```